http_dumper_test.go

```diff
@@ -0,0 +1,250 @@
+	req, err = http.NewRequest(http.MethodGet, server.URL, http.NoBody)
+	require.NoError(t, err)
+	resp, err = client.Do(req)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer resp.Body.Close()
+
+	output = stripANSI(buf.String())
+	t.Logf("Dump with debug enabled:\n%s", output)
+	assert.Contains(t, output, "Transaction =>")
+}
+
+// roundTripFunc lets us use a function as a RoundTripper in tests.
+type roundTripFunc func(*http.Request) (*http.Response, error)
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+var ErrSimulatedTransportFailure = errors.New("simulated transport failure")
+
+func TestHTTPDebugTransport_PassThroughRoundTripError(t *testing.T) {
+	transport := NewHTTPDebugTransport(roundTripFunc(func(req *http.Request) (*http.Response, error) {
+		return nil, ErrSimulatedTransportFailure
+	}))
+	transport.SetDebug(false)
+
+	// Create a client with the debug transport
+	client := &http.Client{Transport: transport}
+
+	req, err := http.NewRequest(http.MethodGet, "http://example.invalid", http.NoBody)
+	require.NoError(t, err)
+
+	// Simulate a pass-through failure
+	resp, err := client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "HTTPDebugTransport: pass-through round trip failed")
+	assert.ErrorIs(t, err, ErrSimulatedTransportFailure)
+
+	require.Nil(t, resp)
+}
+
+func TestHTTPDebugTransport_RequestDumpFailure(t *testing.T) {
+	tp := NewHTTPDebugTransport(http.DefaultTransport)
+	tp.SetDebug(true)
+
+	client := &http.Client{Transport: tp}
+
+	// Malformed request: URL exists but has no Scheme/Host
+	req := &http.Request{
+		Method: http.MethodGet,
+		URL:    &url.URL{},
+		Header: http.Header{},
+	}
+
+	resp, err := client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "HTTPDebugTransport: failed to dump request")
+}
+
+type errorBody struct{}
+
+var errSimulatedBodyReadFailure = errors.New("simulated body read failure")
+
+func (errorBody) Read(p []byte) (int, error) { return 0, errSimulatedBodyReadFailure }
+func (errorBody) Close() error               { return nil }
+
+func TestHTTPDebugTransport_ResponseDumpFailure(t *testing.T) {
+	transport := NewHTTPDebugTransport(roundTripFunc(func(req *http.Request) (*http.Response, error) {
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(errorBody{}),
+		}, nil
+	}))
+	transport.SetDebug(true)
+
+	client := &http.Client{Transport: transport}
+
+	//nolint:noctx // no context needed for this unit test: synthetic request
+	req, err := http.NewRequest(http.MethodGet, "http://example.invalid", http.NoBody)
+	require.NoError(t, err)
+	resp, err := client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "HTTPDebugTransport: failed to dump response")
```

Comment:
```suggestion
	require.ErrorContains(t, err, "HTTPDebugTransport: failed to dump response")
```

Please consider enabling testifylint

Reply:
Sure